elfi/methods/methods.py

```diff
@@ -374,87 +397,57 @@ def __init__(self, model, discrepancy=None, **kwargs):
         """
 
         model, self.discrepancy = self._resolve_model(model, discrepancy)
-        outputs = model.parameters + [self.discrepancy]
-
+        outputs = self._ensure_outputs(outputs, model.parameters + [self.discrepancy])
         super(Rejection, self).__init__(model, outputs, **kwargs)
 
-    def sample(self, *args, **kwargs):
-        """
-        Parameters
-        ----------
-        n_samples : int
-            Number of samples to generate from the (approximate) posterior
+    def init_inference(self, n_samples, threshold=None, p=None, n_sim=None):
```

Comment:
I'd prefer quantile, as that's 0..1, whereas percentile is in percents. E.g. "1-percentile" is confusing.

Just `p` is certainly shorter, but `p` is used in many contexts. Does it mean p-value, percentile, probability... or apparently, here, the quantile?